src/vcpkg/base/downloads.cpp

```diff
@@ -982,14 +983,14 @@ namespace vcpkg
                         if (maybe_success)
                         {
                             fs.rename(download_path_part_path, download_path, VCPKG_LINE_INFO);
+                            msg::println(msgDownloadSuccesful, msg::path = download_path.filename());
                             return urls[0];
                         }
-
-                        errors.push_back(std::move(maybe_success).error());
```

Comment:
This function is being fairly careful to put the errors in `errors` rather than emitting them here directly; can you explain why this case is different than the others?

Reply:
I believe this error occurs when there is a sha mismatch. I'm printing rather than putting in errors so that the order of the failure messages makes sense and is consistent with the download failure messages below.

Reply:
I guess I would really like to see at least a comment here explaining what the criteria for 'errors' vs being printed directly is, since changing which is which on and ad-hoc basis like this seems likely to create as many bugs as solved every time it is touched.

However, this function is already awful about that so not going to block over this.